File: frontend/src/types/insider.ts
export interface InsiderTransaction {
  insider_name: string | null;
  insider_title: string | null;
  transaction_type: string;
  shares: number | null;
  price_per_share: number | null;
  transaction_value: number | null;
  transaction_date: string;
}

export interface InsiderActivity {
  insider_score: number | null;
  window_days: number;
  transactions: InsiderTransaction[];
}

File: backend/app/schemas/insider.py
"""Insider activity schemas for API response."""

from datetime import date

from pydantic import BaseModel


class InsiderTransactionResponse(BaseModel):
    insider_name: str | None
    insider_title: str | None
    transaction_type: str
    shares: float | None
    price_per_share: float | None
    transaction_value: float | None
    transaction_date: date

    model_config = {"from_attributes": True}


class InsiderActivityResponse(BaseModel):
    insider_score: float | None
    window_days: int
    transactions: list[InsiderTransactionResponse]
